JiriLojda/kontent-ai-new-sample-app
# KCL-11649 - Centralize smart link instance to context. provide smart …

### Motivation

When working on the sample site, I faced a problem of having multiple places with `useSmartLink` and not being sure if I could spin up more instances of SmartLink SDK.

I sometimes even got error saying something about SmartLink SDK being destroyed already. Unfortunately cannot reproduce reliably.

I reorganized SmartLink initialization code the following way:
* Moved creation of the SmartLink SDK as a singleton to the top of the application, into `SmartLinkProvider` and exposed the SmartLink SDK instance via context
* Created two handy hooks to access it
  * `useSmartLink` - Gives access to the SmartLink SDK singleton
  * `useSmartLinkRefresh` - Attaches the refresh callback to the SmartLink SDK, greatly reducing the boilerplate to handle page refresh (used on the Home page)

### Checklist

- [+] Code follows coding conventions held in this repo
- [+] Tests are passing
- [+] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### How to test

When previewing in Web spotlight, the SmartLink actions should work (edit element / auto-refresh)
